File: app/src/main/java/com/bsas/androiddevs/manejoerrores/ui/dialog/GenericWarningDialog.java
package com.bsas.androiddevs.manejoerrores.ui.dialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.DialogFragment;
import android.support.v4.app.FragmentActivity;
import android.view.View;
import android.widget.TextView;

import com.bsas.androiddevs.manejoerrores.R;

public class GenericWarningDialog extends DialogFragment {

    private int messageResource;

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        FragmentActivity fragmentActivity = this.getActivity();

        View view = fragmentActivity.getLayoutInflater().inflate(R.layout.generic_warning_dialog, null);

        ((TextView) view.findViewById(R.id.alert_message_text)).setText(fragmentActivity.getText(this.messageResource));

        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(fragmentActivity);
        alertDialogBuilder.setView(view);

        return alertDialogBuilder.create();
    }

    public void setMessageResource(int messageResource) {
        this.messageResource = messageResource;
    }
}

File: app/src/main/java/com/bsas/androiddevs/manejoerrores/ui/dialog/GenericWarningDialog.kt
package com.bsas.androiddevs.manejoerrores.ui.dialog

import android.app.Dialog
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.bsas.androiddevs.manejoerrores.R

class GenericWarningDialog(val messageResource: Int) : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val activity = this.activity!!

        val view = activity.layoutInflater.inflate(R.layout.generic_warning_dialog, null)

        view.findViewById<TextView>(R.id.alert_message_text).text = activity.getText(this.messageResource)

        val alert = AlertDialog.Builder(activity)
        alert.setView(view)

        return alert.create()
    }


}